Source/JavaScriptCore/wasm/WasmOMGIRGenerator32_64.cpp

```diff
@@ -4349,8 +4349,15 @@ auto OMGIRGenerator::addSIMDLoadPad(SIMDLaneOperation op, ExpressionType pointer
 Value* OMGIRGenerator::loadFromScratchBuffer(unsigned& indexInBuffer, Value* pointer, B3::Type type)
 {
     unsigned valueSize = m_proc.usesSIMD() ? 2 : 1;
+    ASSERT(!m_proc.usesSIMD());
     size_t offset = valueSize * sizeof(uint64_t) * (indexInBuffer++);
     RELEASE_ASSERT(type.isNumeric());
+    if (type.kind() == B3::TypeKind::Int64) {
+        size_t offsetHi = valueSize * sizeof(uint64_t) * (indexInBuffer++);
+        auto* lo = append<MemoryValue>(m_proc, Load, Int32, origin(), pointer, offset);
+        auto* hi = append<MemoryValue>(m_proc, Load, Int32, origin(), pointer, offsetHi);
+        return append<Value>(m_proc, Stitch, type, origin(), lo, hi);
+    }
```

Comment:
Hmm, I don't immediately see how this can work when `loadFromScratchBuffer` is used in the OSR path? Doesn't this introduce a mismatch between the values that the lower tier saves and the values that we restore (i.e. we expect Int64s broken up in two parts and stored in two 64-bit slots) but the lower tier will just store an Int64 in a single 64-bit slot?

Reply:
Yeah, I missed this. I added a test